autorel/clients/base.py

```diff
@@ -0,0 +1,109 @@
+import requests
+import json
+import sys
+
+__pyversion__ = sys.version_info[0]
+if __pyversion__ == 2: # Python2
+    from urlparse import urljoin
+else: # Python3
+    from urllib.parse import urljoin
+
+DEFAULT_HEADERS = {
+    'User-Agent' : 'syslog-ng-autorel',
+    'From' : 'ankprashar@gmail.com',
+    "Origin": 'black-perl.in'
+}
+
+DEFAULT_TIMEOUT = 30
+
+class BaseClient(object):
+    '''
+        A Base HTTP client
+    '''
+    http_methods = (
+        'head',
+        'get',
+        'post',
+        'put',
+        'patch',
+        'delete',
+    )
+
+    def __init__(self,endpoint,port,timeout=None):
+        '''
+            The Constructor method for a generic client. 
+        '''
+        self._endpoint = endpoint
+        self._port = port
+        self._timeout = timeout if timeout else DEFAULT_TIMEOUT
+        self._headers = DEFAULT_HEADERS
+
+    def carryRequest(self,method='GET',url='',headers={},params={},payload={}):
+        '''
+        Carry out a {method} request.
+        - The url can be a FQDN( Fully qualified Performmain name ) but should have netloc same as that
+          of API endpoint.
+        '''
+        method = method.lower()
+        methodToCall = getattr(requests,method)
```

Comment:
Yeah, sure. Actually I added some more sugar by adding separate methods `.get()` etc so that `carryRequest` is not directly called, but they are unused for now. The following fixes it:
https://github.com/black-perl/syslog-ng-autorel/commit/6787bee3614840dbf02d0abe11562f803fc7a5a0